java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -218,7 +218,13 @@ static long initViewHandle(DType type, int numRows, int nullCount,
         od, vd, nullCount, numRows, childHandles);
   }
 
-  static ColumnVector fromViewWithContiguousAllocation(long columnViewAddress, DeviceMemoryBuffer buffer) {
+  /**
+   * Creates a ColumnVector from a native column_view using a contiguous device allocation.
+   *
+   * @param columnViewAddress address of the native column_view
+   * @param buffer device buffer containing the data referenced by the column view
+   */
+  public static ColumnVector fromViewWithContiguousAllocation(long columnViewAddress, DeviceMemoryBuffer buffer) {
```

Comment:
What is the role that a "contiguous allocation" plays here? I see that it is just creating a ColumnVector from device buffer and nothing else, so not sure what I'm missing here.

Reply:
Contiguous allocation meaning all of the underlying buffers from this column view (i.e.: validity, offsets, data, and recursively from child columns) comes from this single buffer.  Normally these things are allocated separately and thus are not from a single buffer.  The way contiguous tables work in Java is that we build up ColumnVector instances from views (zero-copy, so not like column vs. column_view in native) that reference the underlying buffer (i.e.: via incRefCount).  The buffer is then closed, decrementing the refcount, but it stays allocated because of all the columns referencing it.  When the last column referencing the buffer is finally closed, the buffer is closed when its refcount goes to zero.

So we need the buffer in this method for the reference semantics, and that buffer is the contiguous allocation holding this column (and probably others).